src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,180 +1,165 @@
+      venue?.rooms
+        ?.filter(
+          (room) =>
+            room.title.toLowerCase().includes(searchQuery) &&
+            room.type !== RoomTypes.unclickable
+        )
+        .map((room, index) => (
+          <NavSearchResult
+            key={`room-${room.title}-${index}`}
+            title={room.title}
+            description="Room"
+            image={room.image_url}
+            onClick={() => {
+              setSelectedRoom(room);
+              clearSearchQuery();
+            }}
+          />
+        )) ?? []
     );
+  }, [searchQuery, venue, clearSearchQuery]);
 
-    const venueEventsResults = venueEvents.filter((event) =>
-      event.name.toLowerCase().includes(normalizedSearchQuery)
-    );
+  const foundUsers = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
+    }
 
-    const roomsResults: Room[] =
-      venue?.rooms?.filter(
-        (room) =>
-          room.title.toLowerCase().includes(normalizedSearchQuery) &&
-          room.type !== RoomTypes.unclickable
-      ) ?? [];
+    return worldUsers
+      .filter((user) => user.partyName?.toLowerCase().includes(searchQuery))
+      .map((user) => (
+        <NavSearchResult
+          key={`user-${user.id}`}
+          title={user.partyName as string}
+          image={user.pictureUrl}
+          isAvatar={true}
+          onClick={() => {
+            openUserProfileModal(user);
+            clearSearchQuery();
+          }}
+        />
+      ));
+  }, [searchQuery, worldUsers, clearSearchQuery, openUserProfileModal]);
+
+  const foundEvents = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
+    }
 
-    setSearchResult({
-      rooms: roomsResults,
-      users: venueUsersResults,
-      events: venueEventsResults,
-    });
-  }, [searchQuery, venue, venueEvents, worldUsers]);
+    return venueEvents
+      .filter((event) => event.name.toLowerCase().includes(searchQuery))
+      .map((event) => {
+        // @debt event can be related to the current venue as well but we don't receive its icon
+        const room = venue?.rooms?.find((room) => room.title === event.room);
+        return (
+          <NavSearchResult
+            key={`event-${event.id ?? event.name}`}
+            title={event.name}
+            description={`Event - ${uppercaseFirstChar(
+              formatUtcSecondsRelativeToToday(event.start_utc_seconds)
+            )}`}
+            image={room?.image_url}
+          />
+        );
+      });
+  }, [searchQuery, venueEvents, venue]);
 
   const numberOfSearchResults =
-    searchResult.rooms.length +
-    searchResult.events.length +
-    searchResult.users.length;
-
-  const clearSearchQuery = useCallback(() => {
-    setSearchQuery("");
-  }, []);
+    foundRooms.length + foundEvents.length + foundUsers.length;
 
   const clearSearchIcon = (
     <img
-      className="nav__clear-search"
+      className="NavSearchBar__clear-search"
       src="/icons/nav-dropdown-close.png"
```

Comment:
[NB] It would be good to move this `/icons/nav-dropdown-close.png` file out of `public/` and into `src/assets/icons` or similar, then import it within this file:

- https://create-react-app.dev/docs/adding-images-fonts-and-files/

That way we get proper 'compile time' checks that it exists, and generally makes things easier/better.

You would also want to check if anywhere else is using this same 'magic string' currently too, and be sure to update them as well.

Reply:
I haven't moved icon, but copied it so far.  https://github.com/sparkletown/sparkle/pull/1328/commits/13dbc327f1e0f6b36487ccca3cecf17b2d67c692

Because the icon seems to be used in the `UserSearchBar` component (scss). But if I'm not wrong, the component itself is not used anywhere. So if we decide to remove the component, I can remove the source icon in the `public/` directory as well. I've also mentioned that in your PR. https://github.com/sparkletown/sparkle/pull/1323#discussion_r617548442

Reply:
It seems unused to me as well. I think it's a legacy hangover from old templates that just wasn't cleaned up. You can remove it in this PR, and then we can get rid of the old file here as well.

Reply:
removed the component: https://github.com/sparkletown/sparkle/pull/1328/commits/2574d407862199372cf770036bc25c76283e698d